tests/plugins/balloonpanel/positioning.js

```diff
@@ -80,19 +80,18 @@
 				if ( !this._getFrameMethodReplaced ) {
 					// The problem is also window.getFrame().getClientRect() as it returns different results from dashboard and directly.
 					this._getFrameMethodReplaced = true;
-					var orig = this.editor.window.getFrame;
-
-					this.editor.window.getFrame = function() {
-						var ret = orig.call( this );
-
-						if ( ret ) {
-							ret.getClientRect = function() {
-								return { height: 300, width: 300, left: 1, bottom: 643, right: 301, top: 343 };
-							};
-						}
-
-						return ret;
-					};
+					var editor = this.editor,
+						rect = { height: 300, width: 300, left: 1, bottom: 643, right: 301, top: 343 },
+						frame = editor.window.getFrame(),
+						container = frame.getParent();
+
+					sinon.stub( editor.window, 'getFrame' ).returns( frame );
```

Comment:
🤔 First there is

```js
frame = editor.window.getFrame();
```

and then

```js
sinon.stub( editor.window, 'getFrame' ).returns( frame );
```

Is it to make `editor.window.getFrame()` function always return the result of this particular `editor.window.getFrame()` call? Would the result be different when the original function is called in specific tests?

Reply:
I need to alter `frame.getWindow`, see [line 93](https://github.com/ckeditor/ckeditor-dev/pull/2819/files/cb96f15b1abbdcec465efa634201c1842dc071f6#diff-955b377084efa16b3c9bcbde47dadd09R93).
[`window.getFrame()`](https://github.com/ckeditor/ckeditor-dev/blob/master/core/dom/window.js#L91-L94) returns new object with each call, so I need to have access to returned object to override it's `getWindow`.

Reply:
Actually I can simplify it by calling stub on `CKEDITOR.dom.window.prototype.getViewPaneSize`.